extmod/uasyncio: Fix syntax of generator functions

damz opened the pull request:
### What?

Fix the syntax of generator functions in the `uasyncio` package:

* `await` is only valid in an `async` function
* `async` functions that use `yield` are actually async generators (a construct not supported by the compiler right now)

### Why?

The compiler is not picky right now, but these are actually all syntax errors.

dpgeorge commented on extmod/uasyncio/event.py:
```diff
@@ -23,7 +23,7 @@ def set(self):
         self.state = False
 
-    async def wait(self):
+    def wait(self):
```
I think we should add a single-line comment to let users know this is actually an async function (a leaf async function) that needs to be await'd.  Eg:
```py
# async
def wait(self):
    ...
```

dpgeorge replied:
(and please add the same comment to all functions in this PR that had the `async` removed)

damz replied:
Sounds reasonable. Done.

dpgeorge commented:
Thanks for this, I agree with the change.  It's actually close to what I did in #6668, which adds support for async generators.

damz committed:
extmod/uasyncio: Fix syntax of generator functions.

The compiler is not picky right now, but these are actually all syntax
errors:
- await is only valid in an async function
- async functions that use yield are actually async generators (a construct
  not supported by the compiler right now)

damz commented:
@dpgeorge Thanks for the review. I updated the PR as requested.
